test/e2e/util.go

```diff
@@ -0,0 +1,448 @@
+	t.Helper()
+	platform := infraConfig.Status.PlatformStatus
+	if platform == nil {
+		t.Fatalf("platform status is missing for infrastructure %s", infraConfig.Name)
+	}
+	switch platform.Type {
+	case configv1.AWSPlatformType:
+		input := route53.ListResourceRecordSetsInput{
+			HostedZoneId:    &zoneID,
+			MaxItems:        StringRef("10"),
+			StartRecordName: &hostRecord,
+		}
+		output, err := dnsClient.aws.ListResourceRecordSets(&input)
+		if err != nil {
+			return false, err
+		}
+		for len(output.ResourceRecordSets) != 0 {
+			for _, recordset := range output.ResourceRecordSets {
+				if *recordset.Name == hostRecord {
+					return true, nil
+				}
+			}
+			if !(*output.IsTruncated) {
+				return false, nil
+			} else {
+				input.StartRecordName = output.NextRecordName
+				output, err = dnsClient.aws.ListResourceRecordSets(&input)
+				if err != nil {
+					return false, err
+				}
+			}
+		}
+	default:
+		t.Fatalf("Unsupported Provider")
+	}
+	return false, nil
+}
+
+func StringRef(in string) *string {
+	return &in
+}
+
+func BoolRef(in bool) *bool {
+	return &in
+}
+
+func Int64Ref(in int64) *int64 {
+	return &in
+}
+
+func defaultExternalDNS(t *testing.T, zoneID string, credsSecret *corev1.Secret) operatorv1alpha1.ExternalDNS {
+	t.Helper()
+	extdns := defaultExtDNSTemplate
+
+	platform := infraConfig.Status.PlatformStatus
+	if platform == nil {
+		t.Fatalf("platform status is missing for infrastructure %s", infraConfig.Name)
+	}
+	var provider operatorv1alpha1.ExternalDNSProvider
+	switch platform.Type {
+	case configv1.AWSPlatformType:
+		provider = operatorv1alpha1.ExternalDNSProvider{
+			Type: operatorv1alpha1.ProviderTypeAWS,
+			AWS: &operatorv1alpha1.ExternalDNSAWSProviderOptions{
+				Credentials: operatorv1alpha1.SecretReference{
+					Name: credsSecret.Name,
+				},
+			},
+		}
+	default:
+		t.Fatalf("Unsupported Provider")
+	}
+
+	extdns.Spec.Provider = provider
+	extdns.Spec.Zones = []string{zoneID}
+
+	extdns.Spec.Source.FQDNTemplate = []string{fmt.Sprintf("{{.Name}}.%s", domainName)}
+
+	return extdns
+}
+
+func createCredsSecret(t *testing.T, cl client.Client) (*corev1.Secret, error) {
+	t.Helper()
+	originalCredsSecret := &corev1.Secret{}
+	originalCredsSecretName := types.NamespacedName{
+		Name:      "aws-creds",
+		Namespace: "kube-system",
+	}
+	if err := cl.Get(context.TODO(), originalCredsSecretName, originalCredsSecret); err != nil {
+		t.Logf("failed to get credentials secret %s: %v", originalCredsSecretName.Name, err)
+		return nil, err
+	}
+
+	credsSecret := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("aws-access-key-%s", randomURLString(16)),
+			Namespace: "external-dns-operator",
+		},
+		Data: map[string][]byte{
+			"aws_access_key_id":     originalCredsSecret.Data["aws_access_key_id"],
+			"aws_secret_access_key": originalCredsSecret.Data["aws_secret_access_key"],
+		},
+	}
+	if err := cl.Create(context.TODO(), &credsSecret); err != nil {
```

Comment:
The external dns operator syncs the secret mentioned in `provider.aws.credentials` into the operator's namespace. I guess that this may be enough to just reference `kube-system/aws-creds` in ExternalDNS CR.

Reply:
That makes sense. I'll try this in a future iteration and simplify it if it works